### PR TITLE
Add a test_on_native_only option

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by scopatz** [comment]
Please add documentation for this new parameter: https://conda-forge.org/docs/maintainer/conda_forge_yml.html

File: news/test_on_native_only.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+**Added:**
+
+* Add an option `test_on_native_only` to not run tests when cross compiling
```
Comment (scopatz) [inline news/test_on_native_only.rst:3]: double backticks needed

**Comment by isuruf** [comment]
Sure. I'll send a PR after this is merged.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**